addons/web_editor/static/src/js/wysiwyg/widgets/media.js

```diff
@@ -1364,55 +1572,162 @@ var VideoWidget = MediaWidget.extend({
+                        const embedURL = (videoData && !videoData.error) ? videoData.embed_url.split("?")[0] : false;
+                        let currentURL;
+                        let foundIndex;
+                        for (const att of this.attachments) {
+                            currentURL = att.url ? att.url.split("?")[0] : false;
+                            if (currentURL && embedURL && currentURL === embedURL) {
+                                attachment = [att];
+                                foundIndex = this.attachments.indexOf(att);
+                            }
+                        }
+                        if (attachment.length !== 0 && foundIndex !== -1) {
+                            if (foundIndex >= this.numberOfAttachmentsToDisplay) {
+                                this.previousAttachment = attachment[0];
+                                await this._renderThumbnails();
+                            }
+                            await this._selectAttachment(attachment[0], false);
+                        }
+                        if (!attachment && embedURL) {
+                            let searchDomain = this._getAttachmentsDomain("");
+                            searchDomain.push(["url", "=like", `%${embedURL}%`]);
+                            // Attempt to fetch it from DB
+                            attachment = await this._rpc({
+                                model: 'ir.attachment',
+                                method: 'search_read',
+                                args: [],
+                                kwargs: {
+                                    domain: searchDomain,
+                                    fields: ['name', 'mimetype', 'description', 'checksum', 'url', 'type', 'res_id', 'res_model', 'public', 'access_token', 'src', 'width', 'height', 'original_id', 'thumbnail'],
+                                    order: [{name: 'id', asc: false}],
+                                    context: this.options.context,
+                                    limit: 1,
+                                    offset: 0,
+                                }
+                            });
+                            if (attachment.length !== 0) {
+                                this.previousAttachment = attachment[0];
+                                await this._renderThumbnails();
+                                await this._selectAttachment(attachment[0], false);
+                            }
+                        }
+                    }
+                }
+                // Toggle URL validation classes
+                this.$el.find('#o_video_form_group')
+                    .toggleClass('o_has_error', !query.$video).find('.form-control, .custom-select').toggleClass('is-invalid', !query.$video)
+                    .end()
+                    .toggleClass('o_has_success', !!query.$video).find('.form-control, .custom-select').toggleClass('is-valid', !!query.$video);
```

Comment:
A bit hard to read :hot_face: , and very long lines
```suggestion
                const $formGroup = this.$el.find('#o_video_form_group');
                $formGroup.toggleClass('o_has_error', !query.$video)
                          .toggleClass('o_has_success', !!query.$video);
                $formGroup.find('.form-control, .custom-select')
                          .toggleClass('is-valid', !!query.$video)
                          .toggleClass('is-invalid', !query.$video);
```